dev/generate_artifacts.jl

```diff
@@ -1,69 +1,83 @@
+using Base: SHA1
+using JSON: JSON
+using HTTP: HTTP
+using Pkg.GitTools: tree_hash
 using Pkg.Artifacts
-using TimeZones.TZData: tzdata_url
-using SHA
-using Pkg.BinaryPlatforms
-using TimeZones.TZData: extract, tzdata_download
+using SHA: sha256
+using TimeZones.TZData: extract, tzdata_url, tzdata_versions
 
-# This is the path to the Artifacts.toml we will manipulate
-artifacts_toml = joinpath(dirname(@__DIR__), "Artifacts.toml")
+# Code loosely based upon: https://julialang.github.io/Pkg.jl/dev/artifacts/#Using-Artifacts-1
+function bind_artifact_url!(artifacts_toml::String, name::String, url::String)
+    # Query the `Artifacts.toml` file for the hash associated with artifact name. If no such
+    # binding exists within the file then `nothing` will be returned.
+    artifact_hash = Artifacts.artifact_hash(name, artifacts_toml)
 
-const VERSIONS = let
-    file = download("https://data.iana.org/time-zones/releases/")
-    html = read(file, String)
-    rm(file)
-    [m[:version] for m in eachmatch(r"href=\"tzdata(?<version>(?:\d{2}){1,2}[a-z]?).tar.gz\"", html)]
-end
+    if artifact_hash === nothing
+        @info "Processing new artifact: $name"
 
-for version in VERSIONS
-    # Query the `Artifacts.toml` file for the hash bound to the specific version
-    # (returns `nothing` if no such binding exists)
-    tzarchive_latest_hash = artifact_hash("tzdata$version", artifacts_toml)
+        archive = download(url)
+        try
+            # Compute the `sha256` of the archive.
+            archive_sha = bytes2hex(open(sha256, archive))
 
-    # If the name was not bound, or the hash it was bound to does not exist, create it!
-    if isnothing(tzarchive_latest_hash) || !artifact_exists(tzarchive_latest_hash)
```

Comment:
Did we stop dealing with the case where the artifact hash is not nothing but it doesn't exist?

Reply:
`artifact_exists` checks that an artifact is present in `~/.julia/artifacts`. Part of the changes done here were to avoid populating this `~/.julia/artifacts` unnecessarily when generating the `Artifacts.toml` I needed to remove the `artifact_exists` check and only check if the artifact was listed in the TOML file which is done via `artifact_hash(...) !== nothing`